lsat/interceptor_test.go

```diff
@@ -300,18 +305,51 @@ func TestInterceptor(t *testing.T) {
 		overallWg.Wait()
 
 		// Interpret result/expectations.
-		if tc.expectToken {
-			if _, err := store.CurrentToken(); err != nil {
-				t.Fatalf("[%s] expected store to contain token",
-					tc.name)
+		assertResultCorrect(t, tc)
+	}
+}
+
+// TestStreamInterceptor tests that the interceptor can handle LSAT protocol
+// responses in streams and pay the token.
+func TestStreamInterceptor(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+
+	// Run through the test cases.
+	for _, tc := range testCases {
```

Comment:
Looks like we could refactor this further into another helper `testInterceptor` and invoke that here with the differences between and `TestUnaryInterceptor` and `TestStreamInterceptor`?